src/middleware/externalRequests.js

```diff
@@ -389,65 +403,81 @@ const sendMappedObject = (
-    Object.keys(request.params).forEach(param => {
-      let parameterValue
-      const queryParam = request.params[`${param}`]
-      const fullPath = queryParam.path
-
-      // remove first index as this defines the type of param to extract
-      const extractType = fullPath.split('.')[0]
-
-      // remove the extractType property from path
-      const path = fullPath.replace(`${extractType}.`, '')
-
-      switch (extractType) {
-        case 'payload':
-          parameterValue = extractValueFromObject(ctx.request.body, path)
-          break
-        case 'query':
-          parameterValue = extractValueFromObject(ctx.request.query, path)
-          break
-        case 'state':
-          parameterValue = extractValueFromObject(ctx.state.allData.state, path)
-          break
-        case 'lookupRequests':
-          parameterValue = extractValueFromObject(
-            ctx.state.allData.lookupRequests,
-            path
-          )
-          break
-        case 'responseBody':
-          parameterValue = extractValueFromObject(
-            ctx.state.allData.responseBody,
-            path
-          )
-          break
-        default:
-          ctx.statusCode = 500
-          throw new Error(
-            `Unsupported Query Parameter Extract Type: ${extractType}`
-          )
-      }
+  if (request.params && request.params.query) {
+    Object.keys(request.params.query).forEach(paramName => {
+      const queryParamOptions = request.params.query[`${paramName}`]
+      const fullPath = queryParamOptions.path
+
+      const parameterValue = extractParamValue(fullPath, ctx)
 
       if (parameterValue) {
-        const prefix = request.params[`${param}`].prefix
-          ? request.params[`${param}`].prefix
+        const prefix = queryParamOptions.prefix ? queryParamOptions.prefix : ''
+        const postfix = queryParamOptions.postfix
+          ? queryParamOptions.postfix
           : ''
-        const postfix = request.params[`${param}`].postfix
-          ? request.params[`${param}`].postfix
-          : ''
-        requestQueryParams[`${param}`] = `${prefix}${parameterValue}${postfix}`
+        requestQueryParams[
+          `${paramName}`
+        ] = `${prefix}${parameterValue}${postfix}`
       }
     })
   }
 
   return requestQueryParams
 }
 
+const resolveRequestUrl = (ctx, request) => {
+  let url = request.url
+
+  if (request.params && request.params.url) {
+    Object.keys(request.params.url).forEach(paramName => {
+      const urlParamOptions = request.params.url[paramName]
+      const fullPath = urlParamOptions.path
+
+      const parameterValue = extractParamValue(fullPath, ctx)
+
+      if (parameterValue) {
+        const prefix = urlParamOptions.prefix ? urlParamOptions.prefix : ''
+        const postfix = urlParamOptions.postfix ? urlParamOptions.postfix : ''
+        url = url.replace(
+          new RegExp(`:${paramName}`, 'g'),
+          `${prefix}${parameterValue}${postfix}`
+        )
+      }
```

Comment:
@rcrichton please could you remove the param from the URL if it does not resolve. If we send through the param template name it could cause issues. 
(I need this logic to distinguish between endpoints)

```suggestion
      } else {
        url = url.replace(new RegExp(`:${paramName}`, 'g'), '')
      }
```